packages/api-gateway/src/modules/user/user.controller.ts

```diff
@@ -124,7 +131,13 @@ export class UserController implements OnModuleInit {
     status: HttpStatus.INTERNAL_SERVER_ERROR,
     description: 'Internal server error.',
   })
-  async setUserType(@Body() setUserTypeParams: SetUserTypeModel) {
+  async setUserType(
+    @User() user: UserProto.User,
+    @Body() setUserTypeParams: SetUserTypeModel,
+  ) {
+    if (user.type !== UserProto.UserType.SUPERADMIN) {
```

Comment:
We should have documentation for the require user types--otherwise someone could get confused why it isn't working when they're not SUPERADMIN